definitions/tools/vcf_sanitize.cwl

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Sanitize a VCF"
+baseCommand: ["/bin/bash","sanitize.sh"]
+requirements:
+    - class: ResourceRequirement
+      ramMin: 4000
+      coresMin: 1
+    - class: DockerRequirement
+      dockerPull: "mgibio/samtools-cwl:1.0.0"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'sanitize.sh'
+        entry: |
+            set -eou pipefail
+
+            # remove lines containing non ACTGN bases, as they conflict with the VCF spec
+            # and cause GATK to choke
+            gunzip -c "$1" | perl -a -F'\t' -ne 'print $_ if $_ =~ /^#/ || $F[3] !~ /[^ACTGNactgn]/' >sanitized.vcf
+            /opt/htslib/bin/bgzip sanitized.vcf
+            tabix sanitized.vcf.gz
+inputs:
+    vcf:
+        type: File
+        inputBinding:
+            position: 1
+outputs:
+    sanitized_vcf:
+        type: File
+        outputBinding:
+            glob: "sanitized.vcf.gz"
```

Comment:
What do you think of having the output file be named `$(inputs.vcf.nameroot).sanitized.vcf.gz`?

Reply:
Is there a straightforward way to strip the full `.vcf.gz` suffix?   `$(inputs.vcf.nameroot)`seems only to strip the final extension `.gz`.

Reply:
Ah yeah, the extension can have at most one `.` per the docs... Options are probably to split with JS or pass the whole basename into the shell script and let it staple the `sanitized` into the final filename.